File: solutions/java/cars-assemble/1/src/main/java/CarsAssemble.java
public class CarsAssemble {

    public double productionRatePerHour(int speed) {
        final int PRODUCTION = 221;

        double factor;
        if (speed == 0) {
            factor = 0;
        } else if (speed <= 4) {
            factor = 1;
        } else if (speed <= 8) {
            factor = 0.9;
        } else if (speed <= 9) {
            factor = 0.8;
        } else {
            factor = 0.77;
        }

        return factor * speed * PRODUCTION;
    }

    public int workingItemsPerMinute(int speed) {
        return (int) productionRatePerHour(speed) / 60;
    }
}

File: solutions/csharp/cars-assemble/1/CarsAssemble.cs
using System;
using System.ComponentModel;

static class AssemblyLine
{
    public static double SuccessRate(int speed)
    {
        if (speed < 0 || speed > 10)
        {
            throw new ArgumentException("speed must be in range 1 .. 10");
        }

        double rate = 0.0;
        if (speed == 0)
            rate = 0.0;
        else if (speed <= 4)
            rate = 1.0;
        else if (speed <= 8)
            rate = 0.9;
        else if (speed == 9)
            rate = 0.8;
        else rate = 0.77;

        return rate;
    }
    
    public static double ProductionRatePerHour(int speed)
    {
        return speed * SuccessRate(speed) * 221;
    }

    public static int WorkingItemsPerMinute(int speed)
    {
        return (int)(ProductionRatePerHour(speed) / 60);
    }
}
